modules/memory-management/src/main/java/org/mule/runtime/internal/memory/bytebuffer/ByteBufferProviderBuilder.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.internal.memory.bytebuffer;
+
+import org.mule.runtime.api.memory.provider.ByteBufferPoolConfiguration;
+import org.mule.runtime.api.memory.provider.ByteBufferProvider;
+import org.mule.runtime.api.memory.provider.type.ByteBufferType;
+
+import java.nio.ByteBuffer;
+
+public class ByteBufferProviderBuilder {
+
+  private final boolean isDirect;
+
+  private ByteBufferPoolConfiguration poolConfiguration;
+
+  private int maxBufferSize;
```

Comment:
Shouldn't this size be part of the poolConfiguration?
If not, the "if" in the build only considers the nullity of poolConfiguration